packages/Kokkos/test/tstDetailsTreeConstruction.cpp

```diff
@@ -40,45 +40,56 @@ TEUCHOS_UNIT_TEST( DetailsBVH, morton_codes )
         ref[i] = fun( anchors[i] );
     // using points rather than boxes for convenience here but still have to
     // build the axis-aligned bounding boxes around them
-    std::vector<dtk::Box> boxes( n );
+    using DeviceType = typename NO::device_type;
+    Kokkos::View<dtk::Box *, DeviceType> boxes( "boxes", n );
     for ( int i = 0; i < n; ++i )
         dtk::expand( boxes[i], points[i] );
 
     dtk::Box scene;
-    dtk::calculateBoundingBoxOfTheScene( boxes.data(), n, scene );
+    DataTransferKit::TreeConstruction<NO> tc;
+    tc.calculateBoundingBoxOfTheScene( boxes, scene );
     for ( int d = 0; d < 3; ++d )
     {
         TEST_EQUALITY( scene[2 * d + 0], 0.0 );
         TEST_EQUALITY( scene[2 * d + 1], 1024.0 );
     }
 
-    std::vector<unsigned int> morton_codes(
-        n, Kokkos::ArithTraits<unsigned int>::max() );
-    dtk::assignMortonCodes( boxes.data(), morton_codes.data(), n, scene );
+    Kokkos::View<unsigned int *, DeviceType> morton_codes( "morton_codes", n );
     for ( int i = 0; i < n; ++i )
-        TEST_EQUALITY( morton_codes[i], ref[i] );
+        morton_codes[i] = Kokkos::ArithTraits<unsigned int>::max();
+    tc.assignMortonCodes( boxes, morton_codes, scene );
+    TEST_COMPARE_ARRAYS( morton_codes, ref );
 }
 
-TEUCHOS_UNIT_TEST( DetailsBVH, indirect_sort )
+TEUCHOS_UNIT_TEST_TEMPLATE_1_DECL( DetailsBVH, indirect_sort, NO )
 {
     // need a functionality that sort objects based on their Morton code and
     // also returns the indices in the original configuration
 
     // dummy unsorted Morton codes and corresponding sorted indices as reference
     // solution
-    std::vector<unsigned int> k = {2, 1, 4, 3};
+    //
+    using DeviceType = typename NO::device_type;
+    int constexpr n = 4;
+    Kokkos::View<unsigned int *, DeviceType> k( "k", n );
+    k[0] = 2;
+    k[1] = 1;
+    k[2] = 4;
+    k[3] = 3;
```

Comment:
Is it possible to allocate a vector with the initializer_list and wrap a view around it?